bentoml/service/inference_api.py

```diff
@@ -331,4 +330,9 @@ def handle_cli(self, cli_args: Sequence[str]) -> int:
     def handle_aws_lambda_event(self, event):
         inf_task = self.input_adapter.from_aws_lambda_event(event)
         result = next(iter(self.infer((inf_task,))))
-        return self.output_adapter.to_aws_lambda_event(result)
+        out_event = self.output_adapter.to_aws_lambda_event(result)
+        if isinstance(out_event, dict) and "headers" in out_event:
```

Comment:
`if isinstance(out_event, dict):`

Reply:
I guess in some cases there is not a "headers" in output event?

Reply:
But we should avoid injecting `headers` into any output event since not all lambda is HTTP API gateway.